src/Record/TableConnectorValueTransformer.php

```diff
@@ -28,7 +28,33 @@ public function transform(array $normalizedValue, AbstractAttribute $attribute):
         return [
             'locale'  => $normalizedValue['locale'],
             'channel' => $normalizedValue['channel'],
-            'data'    => $normalizedValue['data'],
+            'data'    => $this->filterRows($normalizedValue['data'], $attribute),
         ];
     }
+
+    /**
+     * Filters out unknown codes from table data
+     *
+     * @param array<array<string, mixed>> $normalizedData
+     * @param AbstractAttribute           $attribute
+     *
+     * @return array<array<string, mixed>>
+     */
+    private function filterRows(array $normalizedData, AbstractAttribute $attribute): array
+    {
+        $tableProperty = $attribute->normalize()['table_property'];
+
+        $codeStructure = static function (array $row) {
+            return $row['code'];
+        };
+
+        $allowedCodes = array_flip(array_map($codeStructure, $tableProperty));
+
+        $data = [];
+        foreach ($normalizedData as $dataRow) {
+            $data[] = array_intersect_key($dataRow, $allowedCodes);
```

Comment:
This will give a warning in php 7. And in php 8 a fatal error.

Warning: array_intersect_key(): Expected parameter 1 to be an array, null given

Reply:
`$dataRow` is an array by definition of this bundle and so is `$allowedCodes`. The Bundle currently only supports Akeneo 4's PHP version 7.3. A warning would result in a failing test.


I wasn't 100% sure if phpspec is behaving that way on a warning and tried the following to check the behaviour:

```
    public function it_is_initializable(): void
    {
        trigger_error ('Warning happened', E_USER_WARNING);
        $this->shouldHaveType(TableConnectorValueTransformer::class);
    }
```

The result of the test was:

```
$ vendor/bin/phpspec run
Flagbit/Bundle/ReferenceEntityTableBundle/Record/TableConnectorValueTransformer    
  13  - it is initializable
      warning: Warning happened in
      akeneo-reference-entity-table-bundle/spec/Flagbit/Bundle/ReferenceEntityTableBundle/Record/TableConnectorValueTransformerSpec.php
      line 15

                                      98%                                        96
16 specs
96 examples (95 passed, 1 broken)
132ms
```

1 test was broken. We can depend that any future PHP version, that will cause a warning with the current code, will be covered. The tests even break when notices occure.

---

I also did check the behaviour again how the API reacts on an empty attribute with no columns and a new record. The `TableConnectorValueTransformer` instance isn't invoked in such a case.

Please help me if I missed a spot or code path.

Reply:
Not a real problem when filterRows is used with $normalized['data'] :+1: